backend/src/Edna/Web/API.hs

```diff
@@ -32,3 +44,159 @@ type EdnaAPI =
 
 ednaAPI :: Proxy EdnaAPI
 ednaAPI = Proxy
+
+-- | Endpoints necessary to implement file uploading.
+data FileUploadEndpoints route = FileUploadEndpoints
+  { -- | Parse the file and return its summary for preview.
+    fueParseFile :: route
+      :- "file"
+      :> "parse"
+      :> Summary "Parse the file and return its summary for preview"
+      :> MultipartForm Mem (MultipartData Mem)
+      :> Get '[JSON] FileSummary
+
+  , -- | Upload the file with some methodology and project.
+    fueUploadFile :: route
+      :- "file"
+      :> "upload"
+      :> Summary "Upload the file with some methodology and project"
+      :> ReqBody '[JSON] FileUploadInput
+      :> MultipartForm Mem (MultipartData Mem)
+      :> Post '[JSON] FileSummary
+  } deriving stock (Generic)
+
+type FileUploadAPI = ToServant FileUploadEndpoints AsApi
+
+-- TODO: pagination and sorting are just stubs for now (everywhere).
+-- Most likely we will use @servant-util@ to implement them,
+-- but let's do it later.
+
+-- | Endpoints related to projects.
+data ProjectEndpoints route = ProjectEndpoints
+  { -- | Add a new project.
+    peAddProject :: route
+      :- "project"
+      :> Summary "Add a new project"
+      :> ReqBody '[JSON] Project
+      :> Post '[JSON] (WithId Project)
```

Comment:
yes, as Sasha pointed out extra information would be useful here, we need to show it on frontend as it's added

Reply:
You previously [told](https://serokell.slack.com/archives/C01F8AQSHGE/p1615480272097200?thread_ts=1615459098.060600&cid=C01F8AQSHGE) a different thing. But ok, I'll change it to `WithExtra`, it's a trivial change thanks to the current structure of types.